packages/xstate-lit-controller/src/ActorController.ts

```diff
@@ -0,0 +1,45 @@
+import { ReactiveController, ReactiveControllerHost } from 'lit';
+import { ActorRef, Subscription } from 'xstate';
+import { getSnapshot } from './utils';
+
+export class ActorController<TActor extends ActorRef<any, any>>
+  implements ReactiveController {
+  private host: ReactiveControllerHost;
+  private actorRef: TActor;
+  private subscription: Subscription;
+  private _state: TActor extends ActorRef<any, infer TEmitted>
+    ? TEmitted
+    : never;
+
+  constructor(host: ReactiveControllerHost, actorRef: TActor) {
+    this.host = host;
+    this.actorRef = actorRef;
+    this.host.addController(this);
+
+    this._state = getSnapshot(this.actorRef);
+
+    this.subscription = this.actorRef.subscribe((emitted) => {
+      this._state = emitted;
+      this.host.requestUpdate();
+    });
+  }
+
+  get state() {
+    return this._state;
+  }
+
+  get send() {
+    return this.actorRef.send;
+  }
```

Comment:
Tearing the `send()` method off like this will possibly give it the wrong `this` reference. It'll also make inlining by the VM harder. Consider doing:

```ts
  send(event, options) {
    return this.actorRef.send(event, options);
  }
```